### PR TITLE
Improve oracle alter table syntax

### LINKED ISSUE #26979: Support parsing Oracle ALTER TABLE sql
# Background
Hi community,
This issue is for #26878.

ShardingSphere parser engine helps users parse a SQL to get the AST (Abstract Syntax Tree) and visit this tree to get SQLStatement (Java Object). Currently, we are planning to enhance the support for Oracle SQL parsing in ShardingSphere.

More details:
https://shardingsphere.apache.org/document/current/en/reference/sharding/parse/

# Task
This issue is to support more oracle sql parse, as follows:
```sql
ALTER TABLE admin_iot3 ADD OVERFLOW TABLESPACE admin_tbs2;
```

```sql
ALTER TABLE countries_demo ADD OVERFLOW;
```

```sql
ALTER TABLE dept
    ENABLE NOVALIDATE CONSTRAINT dname_ukey;

ALTER TABLE dept
    ENABLE NOVALIDATE PRIMARY KEY,
    ENABLE NOVALIDATE UNIQUE (dname, loc);
```

```sql
ALTER TABLE customers
   ADD (online_acct_pw VARCHAR2(8) ENCRYPT NO SALT 'NOMAC');
```

```sql
ALTER TABLE employees
   ENABLE NOVALIDATE PRIMARY KEY
   ENABLE NOVALIDATE CONSTRAINT emp_last_name_nn;
```

# Process
1. First confirm that this is a correct oracle sql syntax, if not please ignore;
2. Compare SQL definitions in Oficial SQL Doc and ShardingSphere SQL Doc;
3. If there is any difference in ShardingSphere SQL Doc, please correct them by referring to the Official SQL Doc;
4. Run mvn install the current_file_module;
5. Check whether there are any exceptions. If indeed, please fix them. (Especially xxxVisitor.class);
6. Add new corresponding SQL case in SQL Cases and expected parsed result in Expected Statment XML;
7. Run SQLParserParameterizedTest to make sure no exceptions.

# Relevant Skills
1. Master JAVA language
2. Have a basic understanding of Antlr `g4` file
3. Be familiar with Oracle SQLs

### PR DESCRIPTION
**Pull request by zihaoAK47**
Fixes #26979.

Changes proposed in this pull request:
  - Improve oracle alter table syntax
  - Add test sql

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ✓] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ✓] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ✓] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ✓] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 1]
```diff
@@ -295,7 +295,7 @@ identityOption
     ;
 
 encryptionSpecification
-    : (USING STRING_)? (IDENTIFIED BY STRING_)? STRING_? (NO? SALT)?
+    : (USING STRING_)? (IDENTIFIED BY STRING_)? (integrityAlgorithm? (NO? SALT)? | (NO? SALT)? integrityAlgorithm?)
```
Comment (zihaoAK47) [inline parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:298]: Test execution:

SQL> ALTER TABLE mytable ADD (online_acct_pw VARCHAR2(8) ENCRYPT NO SALT 'NOMAC');
ALTER TABLE mytable ADD (online_acct_pw VARCHAR2(8) ENCRYPT NO SALT 'NOMAC')
                         *
ERROR at line 1:
ORA-28336: cannot encrypt SYS owned objects


SQL> ALTER TABLE mytable ADD (online_acct_pw VARCHAR2(8) ENCRYPT 'NOMAC' NO SALT);
ALTER TABLE mytable ADD (online_acct_pw VARCHAR2(8) ENCRYPT 'NOMAC' NO SALT)
                         *
ERROR at line 1:
ORA-28336: cannot encrypt SYS owned objects

File: parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 2]
```diff
@@ -427,7 +427,7 @@ alterTableProperties
     | parallelClause
     | rowMovementClause
     | logicalReplicationClause
-    | flashbackArchiveClause)+ | renameTableSpecification) alterIotClauses? alterXMLSchemaClause?
+    | flashbackArchiveClause)+ | renameTableSpecification)? alterIotClauses? alterXMLSchemaClause?
```
Comment (zihaoAK47) [inline parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:430]: SQL> ALTER TABLE mytable ADD OVERFLOW TABLESPACE admin_tbs2;
ALTER TABLE mytable ADD OVERFLOW TABLESPACE admin_tbs2
            *
ERROR at line 1:
ORA-28658: This operation is supported only for Index-Organized tables

File: parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 3]
```diff
@@ -673,11 +673,11 @@ memOptimizeWriteClause
     ;
 
 enableDisableClauses
-    : (enableDisableClause | enableDisableOthers)?
+    : (enableDisableClause | enableDisableOthers)+
```
Comment (zihaoAK47) [inline parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:676]: https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/ALTER-TABLE.html#GUID-552E7373-BF93-477D-9DA3-B2C9386F2877

SQL> ALTER TABLE mytable ENABLE NOVALIDATE PRIMARY KEY ENABLE NOVALIDATE CONSTRAINT emp_last_name_nn;
ALTER TABLE mytable ENABLE NOVALIDATE PRIMARY KEY ENABLE NOVALIDATE CONSTRAINT emp_last_name_nn
*
ERROR at line 1:
ORA-02430: cannot enable constraint (EMP_LAST_NAME_NN) - no such constraint

File: parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 4]
```diff
@@ -673,11 +673,11 @@ memOptimizeWriteClause
...
     ;
 
 enableDisableClause
-    : (ENABLE | DISABLE) (VALIDATE |NO VALIDATE)? ((UNIQUE columnName (COMMA_ columnName)*) | PRIMARY KEY | constraintWithName) usingIndexClause? exceptionsClause? CASCADE? ((KEEP | DROP) INDEX)?
+    : (ENABLE | DISABLE) (VALIDATE | NOVALIDATE)? ((UNIQUE columnName (COMMA_ columnName)*) | PRIMARY KEY | constraintWithName) usingIndexClause? exceptionsClause? CASCADE? ((KEEP | DROP) INDEX)?
```
Comment (zihaoAK47) [inline parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:680]: https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/ALTER-TABLE.html#GUID-552E7373-BF93-477D-9DA3-B2C9386F2877__I2183600

File: parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 5]
```diff
@@ -3376,7 +3376,7 @@ overflowClause
     ;
 
 addOverflowClause
-    : ADD OVERFLOW segmentAttributesClause? LP_ PARTITION segmentAttributesClause? (COMMA_ PARTITION segmentAttributesClause?)* RP_
+    : ADD OVERFLOW segmentAttributesClause? (LP_ PARTITION segmentAttributesClause? (COMMA_ PARTITION segmentAttributesClause?)* RP_)?
```
Comment (zihaoAK47) [inline parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:3379]: https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/ALTER-TABLE.html#GUID-552E7373-BF93-477D-9DA3-B2C9386F2877__I2082143

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**